socketio/src/client/socket.rs

```diff
@@ -20,6 +22,16 @@ pub struct Socket {
 
 type SocketCallback = dyn FnMut(Payload, Socket) + 'static + Sync + Send;
 
+/// Flavor of engineio transport
+//TODO: Consider longevity
```

Comment:
What do you mean by that?

Reply:
Is storing the underlying transport type in an enum (with those options) what we want to use in the future? Moving or modifying (even adding) the enum in the future is a breaking change.